### ISSUE TITLE
Evaluation: Pearson: nan  Spearman: nan

### ISSUE THREAD

**Issue by MaxWenzel** [UKPLab/sentence-transformers#2702]
I'm finetuning a Cross-Encoder-Model on my own train- and testdata. I use the CECorrelationEvaluator for evaluation during the training. 
The testdata (for evaluation) contains 218021 InputExamples. The labels (scores) are distributed as follows:

label
0.0       7755
0.4     74659
0.8    118280
0.9       4956
1.0       12371

Unfortunately, Pearson and Spearman coefficients are mostly NaN (and if not, they have very small values):

2024-05-31 13:30:33 - CECorrelationEvaluator: Evaluating the model on sts-cross dataset in epoch 2 after 10000 steps:
2024-05-31 13:32:33 - Correlation:	Pearson: -0.0098	Spearman: -0.0072
2024-05-31 13:43:42 - CECorrelationEvaluator: Evaluating the model on sts-cross dataset in epoch 2 after 20000 steps:
2024-05-31 13:45:43 - Correlation:	Pearson: nan	Spearman: nan
2024-05-31 13:56:56 - CECorrelationEvaluator: Evaluating the model on sts-cross dataset in epoch 2 after 30000 steps:
2024-05-31 13:58:56 - Correlation:	Pearson: nan	Spearman: nan
2024-05-31 14:03:54 - CECorrelationEvaluator: Evaluating the model on sts-cross dataset after epoch 2:
2024-05-31 14:05:55 - Correlation:	Pearson: nan	Spearman: nan
2024-05-31 14:17:08 - CECorrelationEvaluator: Evaluating the model on sts-cross dataset in epoch 3 after 10000 steps:
2024-05-31 14:19:10 - Correlation:	Pearson: nan	Spearman: nan
2024-05-31 14:30:29 - CECorrelationEvaluator: Evaluating the model on sts-cross dataset in epoch 3 after 20000 steps:
2024-05-31 14:32:29 - Correlation:	Pearson: nan	Spearman: nan
2024-05-31 14:43:44 - CECorrelationEvaluator: Evaluating the model on sts-cross dataset in epoch 3 after 30000 steps:
2024-05-31 14:45:45 - Correlation:	Pearson: nan	Spearman: nan
2024-05-31 14:50:51 - CECorrelationEvaluator: Evaluating the model on sts-cross dataset after epoch 3:
2024-05-31 14:52:52 - Correlation:	Pearson: nan	Spearman: nan
2024-05-31 15:03:56 - CECorrelationEvaluator: Evaluating the model on sts-cross dataset in epoch 4 after 10000 steps:
2024-05-31 15:05:57 - Correlation:	Pearson: nan	Spearman: nan

Do you have any clue for me where the problem might lie? I know there are already several tickets on this topic. But the problem there mostly was that always the same score was used.

**Comment by MaxWenzel** [comment]
Ok, that's cleared up. I debugged the CECorrelationEvaluator class and saw that my model returns the same score for every sentence pair. This was probably because I had set the learning rate of my Adam optimizer too high (overfitting).

**Comment by tomaarsen** [comment]
Hello!

I'm glad to see that you figured this out! I hadn't gotten around to experimenting with this. Best of luck with your model!

- Tom Aarsen
